### PR TITLE
ARROW-5826: [Website] Blog post for 0.14.0 release announcement

### PR DESCRIPTION
**Pull request by wesm**
I plan to publish by noon US/Eastern time on Monday July 8 -- please post commits with any edits or push directly to my branch and I will include the changes in the post.

### CODE REVIEW

**Review by nevi-me — APPROVED** [review]
No changes from my side on Rust (or other) details. Thanks Wes

File: site/_posts/2019-07-08-0.14.0-release.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+* A proposal for versioning and forward/backward compatibility
+  guarantees for the 1.0.0 release was shared, not much discussion has
+  occurred yet.
+* Addressing possible unaligned access and undefined behavior concerns
+  in the Arrow binary protocol
+* Supporting smaller than 128-bit encoding of fixed width decimals
+* Forking the Avro C++ implementation so as to adapt it to Arrow's
+  needs
+
```
Comment (emkornfield) [inline site/_posts/2019-07-08-0.14.0-release.md]:
```suggestion
*  Sparse representation and compression in Arrow
```

**Comment by kou** [comment]
I've fixed some typos.
I've also added the suggestion by @emkornfield.
I've also added "Packaging" section. I hope that this section is tweaked by native English speakers.

**Comment by wesm** [comment]
@kou I'll review your changes. Thanks!

**Comment by wesm** [comment]
+1

**Pull request closed without merging**